pynbody/plot/sph.py

```diff
@@ -464,13 +471,29 @@ def image(sim, qty='rho', width="10 kpc", resolution=500, units=None, log=True,
             units = im.units
 
 
-        if units.latex() is "":
-            units=""
-        else:
-            units = "$"+units.latex()+"$"
-
+        # RS - Fix formatting.
         if show_cbar:
-             plt.colorbar(ims).set_label(qtytitle+"/"+units)
+            if log:
+                l_f = LogFormatterExponent() # sometimes tacks 'e' on value...???
+            else:
+                l_f = ScalarFormatter()
+
+            if qtytitle is not None:
+                plt.colorbar(ims,format=l_f).set_label(qtytitle)
+            else:
+                if log:
+                    plt.colorbar(ims,format=l_f).set_label("$log\; %s$" % units)
+#                    plt.colorbar(ims,
+                else:
+                    plt.colorbar(ims,format=l_f).set_label(units)
+        
```

Comment:
I am worried about these changes. Please see comments on revision 5acf3411b3d101b4df, which this partially (or fully) reverts. I kept getting really bizarre results on my plots and I think this complicated logic may only work for a small number of cases.